source/appController/AMDSClientAppController.h

```diff
@@ -0,0 +1,90 @@
+#ifndef AMDSCLIENTAPPCONTROLLER_H
+#define AMDSCLIENTAPPCONTROLLER_H
+
+#include <QObject>
+#include <QHash>
+#include <QDateTime>
+#include <QTcpSocket>
+
+class QNetworkSession;
+
+class AMDSServer;
+class AMDSClientTCPSocket;
+class AMDSClientRequest;
+
+#define AMDS_CLIENT_ERR_INVALID_MESSAGE_TYPE 10100
+#define AMDS_CLIENT_INFO_HAND_SHAKE_MESSAGE 10101
+
+#define AMDS_CLIENT_ERR_SERVER_NOT_CONNECTED 10201
+
+class AMDSClientAppController : public QObject
+{
+    Q_OBJECT
+
+public:
+	explicit AMDSClientAppController(QObject *parent = 0);
+	~AMDSClientAppController();
+
+	/// helper function to check whether the network session is openned or not
+	bool isSessionOpen();
+
+	/// helper function to return the server by given identifier
+	inline AMDSServer *getServerByServerIdentifier(QString serverIdentifier) { return activeServers_.value(serverIdentifier, 0);}
```

Comment:
Here again, make sure that your class-based parameters use `const QString &` (or `const Class &` more generally).